src/main/java/it/pagopa/pn/bff/service/senderdashboard/resources/DatalakeS3Resource.java

```diff
@@ -0,0 +1,311 @@
+package it.pagopa.pn.bff.service.senderdashboard.resources;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import it.pagopa.pn.bff.config.PnBffConfigs;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffSenderDashboardDataResponse;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffSenderDashboardDigitalNotificationFocus;
+import it.pagopa.pn.bff.generated.openapi.server.v1.dto.BffSenderDashboardNotificationOverview;
+import it.pagopa.pn.bff.mappers.senderdashboard.DatalakeDigitalNotificationFocusMapper;
+import it.pagopa.pn.bff.mappers.senderdashboard.DatalakeNotificationOverviewMapper;
+import it.pagopa.pn.bff.service.senderdashboard.exceptions.SenderNotFoundException;
+import it.pagopa.pn.bff.service.senderdashboard.model.DatalakeDigitalNotificationFocus;
+import it.pagopa.pn.bff.service.senderdashboard.model.DatalakeNotificationOverview;
+import it.pagopa.pn.bff.service.senderdashboard.model.IndexObject;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.stereotype.Component;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.AwsSessionCredentials;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.GetObjectResponse;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+import software.amazon.awssdk.services.sts.model.AssumeRoleResponse;
+import software.amazon.awssdk.services.sts.model.Credentials;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.*;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class is responsible for managing the S3 client and retrieving objects
+ * from the data-lake S3 bucket for the sender dashboard.
+ */
+@Slf4j
+@Lazy
+@Component
+@RequiredArgsConstructor
+public class DatalakeS3Resource {
```

Comment:
Can we call the mapper inside the service? I think that is better that this client returns the Datalake DTO instead of the Bff one